src/app/pages/home/home.jsx

```diff
@@ -0,0 +1,123 @@
+
+/**
+ * Dependencies.
+ */
+
+var React = require('react');
+var StoreMixin = require('fluxible').StoreMixin;
+var GoogleMapsStore = require('../../maps/google-maps-store');
+var NavLink = require('../../nav/nav-link');
+var loadGoogleMaps = require('../../maps/load-google-maps');
+
+/**
+ * Home page.
+ */
+
+module.exports = React.createClass({
+  mixins: [StoreMixin],
+  statics: {
+    storeListeners: [GoogleMapsStore]
+  },
+  getInitialState: function () {
+    var venues = this.props.context.getStore('venue').getVenues();
+
+    return {
+      venues: venues
+    }
+  },
+  componentDidMount: function () {
+    var mapsIsReady = this.props.context.getStore('google-maps').isReady();
+    if (mapsIsReady) {
+      this.renderMap();
+    } else {
+      loadGoogleMaps(this.props.context);
+    }
+  },
+  render: function() {
+    return (
+      <div id="home-page">
+        <div id="map-canvas" ref="mapCanvas"></div>
+        <div id="list">
+          <ul>
+          {this.state.venues.map(function (venue) {
+            return <li key={venue.id}>
+              {venue.name + ': '} 
+
+              {/*venue.address.line1 +' '+ 
+              venue.address.line2 +' '+ 
+              venue.address.city +', '+ 
+              venue.address.state +' '+ 
+              venue.address.zip +' ['+
+              venue.address.lat +', '+
+              venue.address.lng +']'*/}
+
+              <ul>
+              {venue.performanceSpaces.map(function (space) {
+                return <li key={space.id}>{space.name}</li>;
+              })}
+              </ul>
+            </li>;
+          })}
+          </ul>
+        </div>
+      </div>
+    );
+  },
+  onChange: function () {
+    var mapsIsReady = this.props.context.getStore('google-maps').isReady();
+    if (mapsIsReady) {
+      this.renderMap();
+    }
+  },
+  renderMap: function () {
+    var mapOptions = {
+      center: { lat: 47.614848, lng: -122.3359059},
+      zoom: 12
+    };
+    var map = new google.maps.Map(document.getElementById('map-canvas'), mapOptions);
+
+    try { window.map = map; } catch (e) {} // for test.
+
+    this.map = map;
+    this.mapMarkers = {};
+    this.updatePins();  
+  },
+  updatePins: function () {
+    var venues = this.props.context.getStore('venue').getVenues();
+    if (this.map) {
+
+      // kill dead markers:
+      var stillAliveVenues = {};
+      for (var i = 0; i < venues.length; i++) {
+        stillAliveVenues[venues[i].id] = 1;
+      }
+
+      var keys = Object.keys(this.mapMarkers);
+      var deadOnes = [];
+      for (var i = 0; i < keys.length; i++) {
```

Comment:
Probably needs refactor. I have no way to test this currently, so I... am not going to do it now.